type_lens/type_view.py

```diff
@@ -235,6 +235,8 @@ def is_subclass_of(self, typ: Any | tuple[Any, ...], /) -> bool:
         Returns:
             Whether the annotation is a subclass of the given type(s).
         """
+        if self.is_union:
```

Comment:
This starts to get back into issue raised in https://github.com/litestar-org/type-lens/pull/9, where you didn't want to merge a change that would check `isinstance(typ, type)` inside `is_subtype_of`.

I need `TypeView(Union[set, int]).is_[subtype/subclass]_of((list, set))` to return True ultimately. but the given type is some arbitrary user provided type, which might include e.g. Literal or other instances on which `issubclass` cannot be called.

So that leaves me either needing to add support for unions to `is_subclass_of`, which isn't strictly within its purview imo, or to go back to revising `is_subtype_of` to correctly handle non-type inherited inputs before calling `issubclass`.

In my mind, now that both functions exist, `is_subtype_of` should really just call `is_subclass_of` as an implementation detail rather than calling `issubclass`.

I maintain that future theoretical instances where that check might yield a `False` where some special casing might have instead made it yield `True` is a better overall DX than its current behavior of raising a TypeError on what would otherwise be valid inputs to the function where it **should** be returning `False`.